python-avd/pyavd/_eos_designs/structured_config/underlay/dhcp_servers.py

```diff
@@ -40,36 +59,33 @@ def _subnets(self: AvdStructuredConfigUnderlayProtocol) -> list:
                     and "unnumbered" not in uplink["ip_address"]
                     and get(peer_facts, "inband_ztp")
                 ):
-                    subnet = {
-                        "subnet": str(ip_network(f"{uplink['peer_ip_address']}/{uplink['prefix_length']}", strict=False)),
-                        "ranges": [{"start": str(uplink["ip_address"]), "end": str(uplink["ip_address"])}],
-                        "name": f"inband ztp for {peer}-{uplink['interface']}",
-                        "default_gateway": f"{uplink['peer_ip_address']}",
-                    }
-                    subnets.append(subnet)
-
-        return subnets
-
-    @cached_property
-    def _ipv4_ztp_boot_file(self: AvdStructuredConfigUnderlayProtocol) -> str | None:
-        """Returns the file name to allow for ZTP to CV. TODO: Add inband_ztp_bootstrap_file to schema."""
+                    subnet_item = EosCliConfigGen.DhcpServersItem.SubnetsItem(
+                        subnet=str(ip_network(f"{uplink['peer_ip_address']}/{uplink['prefix_length']}", strict=False)),
+                        name=f"inband ztp for {peer}-{uplink['interface']}",
+                        default_gateway=f"{uplink['peer_ip_address']}",
+                    )
+                    subnet_item.ranges.append_new(start=str(uplink["ip_address"]), end=str(uplink["ip_address"]))
+                    dhcp_server.subnets.append(subnet_item)
+
+    def _set_ipv4_ztp_boot_file(self: AvdStructuredConfigUnderlayProtocol, dhcp_server: EosCliConfigGen.DhcpServersItem) -> None:
```

Comment:
Call this _update_  since you are inplace updating the given DhcpServersItem. _set_  should only be when we set the structured config directly.